archive_utils.py

```diff
@@ -3423,3 +3423,83 @@ def folding_resync(cparams,logger):
         logger.info("Updated PSRDB entry in 'foldings' table, ID = {0}".format(cparams["db_fold_id"]))
 
     return
+
+# function to upload specified pulsar archive data products to PSRDB
+# WIP
+def upload_data_archives(output_dir, cparams, psrname, logger):
+
+    # Note - the functionality of this code is based on the renaming scheme performed by cleanup()
+    # Should these expected outputs change, the conditions of this code should be re-assessed
+
+    logger.info("Uploading decimated data products - Pipeline PID = {0}".format(cparams["pid"]))
+    # update - pid now to be included in the naming structure (type)
+    local_pid = cparams["pid"].lower()
+    files_to_store = []
+    files_to_delete = []
+
+    # identify the decimated path
+    output_dir = str(output_dir)
+    decimated_path = os.path.join(output_dir,"decimated")
+
+    # select the required decimated products and structure for upload
+    decimated_archives = sorted(glob.glob(os.path.join(decimated_path, "*")))
+
+    for arch in decimated_archives:
+        
+        # isolate a single FTS archive in case of multiples
+        # choice is currently arbitrary
+        FTS_archives = []
+
+        if "FTS" in arch:
+            FTS_archives.append(arch)
+        
+    if (len(FTS_archives) > 0):
+
+        # select first file for upload
+        # produce renamed copy of the file in PSRDB friendly format
+        FTS_selectfile = FTS_archives[0]
```

Comment:
Why do you only grab the first one? Are you sure this is the FTS file?

Reply:
For a given observation, there should only be one FTS file. The code was written to cover the edge case of there being multiple files, but even if there were they should be identical, so it doesn't matter which one is picked. Therefore, pick the first one for convenience.

As it happens, I found a different bug in this section of code (`FTS_archives = []` should be outside the loop) so this will be getting overwritten shortly anyway.